iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/extractor/IoTDBDataRegionExtractor.java

```diff
@@ -84,6 +90,29 @@ public IoTDBDataRegionExtractor() {
 
   @Override
   public void validate(PipeParameterValidator validator) throws Exception {
+    // Check whether the pattern is legal
```

Comment:
make it a seperate function.
 
validatePattern(validator
  .getParameters()
  .getStringOrDefault(
     Arrays.asList(EXTRACTOR_PATTERN_KEY, SOURCE_PATTERN_KEY),
     EXTRACTOR_PATTERN_DEFAULT_VALUE));